### ISSUE TITLE
bug(node): Cannot find module 'node:stream' or its corresponding type declarations.

### ISSUE THREAD

**Issue by flyman-nv** [lancedb/lancedb#1713]
### LanceDB version

v0.10.0

### What happened?

I am getting blocked on running the db in a typescript VSCode extension.

I'm using:

* Node: v22.9.0
* OS: Mac Ventura 13.6.9
* LanceDB: ^0.10.0
* Apache Arrow: ^17.0.0

[{
	"resource": "/Users/#####/Desktop/projects/nv-code/node_modules/apache-arrow/builder.d.ts",
	"owner": "typescript",
	"code": "2307",
	"severity": 8,
	"message": "Cannot find module 'node:stream' or its corresponding type declarations.",
	"source": "ts",
	"startLineNumber": 8,
	"startColumn": 29,
	"endLineNumber": 8,
	"endColumn": 29
}]

Note that this is one of many errors loading node modules that are prefixed with `node:`

Steps I tried to resolve:

* Installed apache arrow locally with Brew
* Updated ESbuild to ignore the package
* Updated tsconfig to be as permissive as possible
* Installing `@types/node`

### Are there known steps to reproduce?

1. `npm install @lancedb/lancedb`
2. Import the db into any file: `import * as lancedb from "@lancedb/lancedb";`

**Comment by wjones127** [comment]
How did you generate the extension template?

What seems suspicious is you mention using ESBuild, which is designed for web bundling. LanceDB only runs in Node environments, so I wouldn't think you would use it with ESBuild.

**Comment by flyman-nv** [comment]
Here's my deps if that's helpful:

  "devDependencies": {
    "@types/babel__core": "^7.20.5",
    "@types/glob": "^8.1.0",
    "@types/handlebars": "^4.1.0",
    "@types/inquirer": "^9.0.7",
    "@types/jest": "^29.5.12",
    "@types/lodash": "^4.17.5",
    "@types/node": "^12.20.55",
    "@types/node-fetch": "^2.6.11",
    "@types/uuid": "^9.0.8",
    "@types/vscode": "^1.46.0",
    "@typescript-eslint/eslint-plugin": "^4.14.1",
    "@typescript-eslint/parser": "^4.14.1",
    "eslint": "^7.19.0",
    "jest": "^29.7.0",
    "jest-fetch-mock": "^3.0.3",
    "prettier": "^2.2.1",
    "ts-jest": "^29.1.2",
    "typescript": "^4.1.3",
    "vsce": "^2.15.0"
  },
  "dependencies": {
    "@apache-arrow/ts": "^17.0.0",
    "@lancedb/lancedb": "^0.10.0",
    "chalk": "^5.3.0",
    "glob": "^10.3.10",
    "glob-gitignore": "^1.0.14",
    "globify-gitignore": "^1.0.3",
    "handlebars": "^4.7.8",
    "inquirer": "^9.2.14",
    "lodash": "^4.17.21",
    "node-fetch": "^3.3.2",
    "openai": "^4.47.1",
    "rxjs": "^7.8.1",
    "uuid": "^9.0.1",
    "vectra": "^0.5.5"
  }

**Comment by Xuanwo** [comment]
Hi, this issue is invalid because it is a user environment issue: @types/node v12 is too old to provide the 'node:' module type declarations, and bundling with esbuild (a web bundler) is unsuitable for a Node-only library. I'm closing this now. Feel free to create a new issue if the problem still exists.
